### PR TITLE
Find payments with plugin info

### PR DESCRIPTION
**Pull request by andrenpaes**
Extends Payment's find_* methods, adding the with_plugin_info argument

### CODE REVIEW

File: lib/killbill_client/models/payment.rb [hunk 1]
```diff
@@ -12,16 +12,27 @@ class Payment < PaymentAttributes
       has_custom_fields KILLBILL_API_PAYMENTS_PREFIX, :payment_id
 
       class << self
-        def find_by_id(payment_id, options = {})
+        def find_by_id(payment_id, with_plugin_info = false, options = {})
```
Comment (andrenpaes) [inline lib/killbill_client/models/payment.rb]: Another option here, would be to expect a `with_plugin_info` in the options hash, but I'm trying to keep the code some what consistent with other usages of with_plugin_info:
https://github.com/killbill/killbill-client-ruby/blob/master/lib/killbill_client/models/invoice_payment.rb#L10
https://github.com/killbill/killbill-client-ruby/blob/master/lib/killbill_client/models/payment_method.rb#L10

File: lib/killbill_client/models/payment.rb [hunk 2]
```diff
@@ -12,16 +12,27 @@ class Payment < PaymentAttributes
...
+          if with_plugin_info.is_a? Hash
+            options = with_plugin_info
+            with_plugin_info = false
+          end
```
Comment (andrenpaes) [inline lib/killbill_client/models/payment.rb]: Kind of ugly, but keeps backwards compatibility with old method signature.

**Comment by pierre** [comment]
:+1: for harmonizing the signature.

Looking at our [integration tests](https://github.com/killbill/killbill-integration-tests) and [Kaui](https://github.com/killbill/killbill-admin-ui), these methods don't seem much used (as a matter of fact, I could only find [this usage](https://github.com/killbill/killbill-admin-ui/blob/2e000ca2fb28b228299002a9324b58dba00cce28/app/controllers/kaui/chargebacks_controller.rb#L4)). Maybe it's just simpler to go ahead and break backward compatibility (Kaui has extensive tests, so it should be easy to detect if something broke).

**Comment by andrenpaes** [comment]
Sounds good. I'll make the changes them.

**Comment by andrenpaes** [comment]
@pierre I've made the change.

Since we are breaking compatibility, maybe we can release a "major" version of the gem, so Gemfiles that use something like `'~> 0.21.0'` would be "protected" from this change (I think).

**Comment by pierre** [comment]
I don't think we have to bump the major (we're explicitly on `0.x.y` because the APIs are not stable and I'm not sure we're ready to commit to a `1.0.0` just yet).

That being said, Gemfiles with `'~> 0.21.0'` would be protected if we were to release `0.22.0` (but not those with `'~> 0.21'`).

**Comment by andrenpaes** [comment]
Yeah, that's actually what I meant by "major" :)

**Pull request merged**